python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -993,12 +1054,23 @@ def build_asset_outs(asset_outs: Mapping[str, AssetOut]) -> Mapping[AssetKey, Tu
     return outs_by_asset_key
 
 
-def _make_asset_keys(deps: Optional[Union[Set[AssetKey], Set[str]]]) -> Optional[Set[AssetKey]]:
-    """Convert all str items to AssetKey in the set."""
+def _make_asset_keys(
```

Comment:
nit: might be nice to move all the deps coercing & errors out to this / a helper to keep the constructor reasonable